### PR TITLE
AzureRT S54 PR #2 - Record Debug Messages & Fix Add-AzureVhd

### PR DESCRIPTION
**Pull request by huangpf**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@huangpf__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (phuang). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by huangpf** [comment]
Fix #387

File: src/Common/AzurePSCmdlet.cs [hunk 1]
```diff
@@ -254,7 +257,7 @@ protected bool IsVerbose()
 
         protected new void WriteError(ErrorRecord errorRecord)
         {
-            FlushDebugMessages();
+            FlushDebugMessages(IsDataCollectionAllowed());
```
Comment (ogail) [inline src/Common/AzurePSCmdlet.cs]: no need to pass `IsDataCollectionAllowed()` just use it in `FlushDebugMessages`

File: src/Common/AzurePSCmdlet.cs [hunk 2]
```diff
@@ -362,8 +365,13 @@ protected void SafeWriteOutputPSObject(string typeName, params object[] args)
             WriteObject(customObject);
         }
 
-        private void FlushDebugMessages()
+        private void FlushDebugMessages(bool record = false)
         {
+            if (record)
```
Comment (ogail) [inline src/Common/AzurePSCmdlet.cs]:
```csharp
if (IsDataCollectionAllowed())
{
    RecordDebugMessages();
}
```

**Comment by ogail** [comment]
added a nit

File: src/Common/AzurePSCmdlet.cs (hunk 1 above)
Comment (huangpf) [inline src/Common/AzurePSCmdlet.cs]: You mean by default log everything?

File: src/Common/AzurePSCmdlet.cs (hunk 2 above)
Comment (huangpf) [inline src/Common/AzurePSCmdlet.cs]: ok

File: src/Common/AzurePSCmdlet.cs (hunk 1 above)
Comment (huangpf) [inline src/Common/AzurePSCmdlet.cs]: But we only want to log when there is an exception. Do you mean you want to log everything if data collection is allowed?

File: src/Common/AzurePSCmdlet.cs (hunk 2 above)
Comment (huangpf) [inline src/Common/AzurePSCmdlet.cs]: same above

**Pull request merged**